tiering/ctl.c

```diff
@@ -269,37 +269,37 @@ struct memtier_memory *ctl_create_tier_memory_from_env(char *env_var_string)
         ret = ctl_parse_query(qbuf, &kind, &ratio);
         if (ret != 0) {
             log_err("Failed to parse query: %s", qbuf);
-            goto builder_delete;
+            goto create_memtier_failure;
```

Comment:
Maybe `cleanup_after_failure`? Current name might suggest that you want to create a memtier failure, forcing a failure. Consider making it a separate function instead of a goto.

Reply:
Done